### PR TITLE
Mutants - HNZ1 - A medical and scientific adventure!

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request
So instead of redoing zombies, I'm gonna make and rebalance a new antag type.

TL;DR: Mutants are a more interdepartmental venture, requiring scientific interaction to synthesize the cure to this horrific virus. Medical still needs to actually perform the cures via surgery. Virology for the vaccine

Note: It is not a virus in the traditional sense, it is a custom component I just put together.

Let's break this down into sections: New mutants, HNZ1 virus, RNA extractor, RNA vials, RNA recombinator, HNZ1 cure, science nodes

## New mutants
Mutants are fastish "humans" that have succumb to the HNZ1 virus. They are still alive and human in a sense. They look mostly normal. However don't let that fool you. Their brain has been turned to sludge. Most of their higher thought has been reduced to atoms. However they might still be in there, deep down.

These mutants look almost ljke the person they used to be. Only a few signs of infection in the further stages, much like I Am Legend. 

The way the virus is actually spread has changed, instead of being a gauranteed infection, it now checks if the poor sod can be injected. If they cant - no infection! This part is a major debuff.

## HNZ1 Virus
image
"Human Neurovirus Z-1"

The HNZ1 virus, a more technologically advanced and lore-friendly virus. It's manufactured by Nanotrasen's Bioweapons division. Realistically once this thing gets out, it's very hard to stop the spread of it. However, Nanotrasen do have ways to recombinate the RNA of the virus, thus making it less of a problem to deal with. More on this to come later.

## RNA extractor
image
RNA vial and extractor

The RNA extractor is very important in the fight against zombies, this is used to extract RNA from LIVING zombies, you need to attack them with it and they need to be alive. It fills the inserted vial(assuming there is one) with RNA, this can then be uploaded into the RNA recombinator. A mutant can only have it's RNA extracted every so often, so you can't spam extract the same live mutant! What we at NBD suggest is to capture a few and harvest their RNA over time. Remember, stuns work on these guys!(They don't even need to have a ckey attached).

## RNA Recombinator
image
image

Nanotrasen RNA Recombinator

The RNA recombiator can do one of two things:

Create a cure:
The RNA Recombinator is the pinnacle of NBD, it's made to recombinate virus RNA so efficiently that you only need a slither of RNA from the original virus. Use this machine(and protect it, or make loads of them) to manufacture the HNZ1 cure! It takes a while to recombinate the RNA you put inside it, and can only do one at a time, but it works very well! Make sure you keep it powered too! It increments in percentage every so often, depending on internal components(upgrade them to make it MUCH faster). The basic Nanotrasen Bioweapons guide to defeating this virus is harvesting the RNA from live (braindead)specimens, and using these machines to go go gadget. 

Create more virus:
It can also create more virus with the RNA sample, this is useful if you want to make some test monkes!

## HNZ1 cure
image
"Experimental HNZ1 Cure sample"

This is the final stage in the war against the virus, you can completely cure someone with one of these. Simply slap it into them via surgery and they'll be cured in no time! (hopefully). An added benefit to this is that it provides several minutes of immunity, meaning you cannot pick the virus up again if you're bitten! Brilliant. It halts the virus in it's tracks even if said person has not... bit the dirt yet. Tie the zombie up, insert this in them, then in a few minutes... bam, new human!

## SCIENCE!
image

Here's the techweb system for the varus research tech, you can print as many as you want out! R&D is **integral** to this fight against the muties.

## Admin only event
Adds an admin event so admins can do this ICl'y.

I understand there will be some balance concerns with this, this needs TM'd first, I'll deal with all that.

## Changelog
:cl:
add: New antagonist and virus, HNZ1 mutants! See: https://github.com/Skyrat-SS13/Skyrat-tg/pull/4664
/:cl:

### CODE REVIEW

**Comment by FlamingLily** [comment]
(I mean I'm not gonna lie I think this would be fun as an extremely rare random event, but I can see why it might be better as admin spawn only)

**Comment by H0gberg** [comment]
Agreed, making it a somewhat rare random event would be very cool!

File: modular_skyrat/modules/mutants/code/mutant_species.dm [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+/mob/living/carbon/human/species/mutant
+	race = /datum/species/mutant
+
+/mob/living/carbon/human/species/mutant/infectious
+	race = /datum/species/mutant/infectious
+
+/datum/species/mutant/infectious
+	name = "Mutated Abomination"
+	mutanthands = /obj/item/mutant_hand
+	armor = 50
```
Comment (Elli-Skala) [inline modular_skyrat/modules/mutants/code/mutant_species.dm]: hey so with tg's new change to armor this means that the base armor will make hollow points will do zero damage due to doubling the armor effect to 100

**Comment by Trubbl3** [comment]
Gandalf, what if instead of researching the virus cure, it spawns in the CMO's inventory or his locker as a high risk item that can either reveal the zombie cure tech or even make a zombie virus or something similar but more weaker, making finally a CMO high risk item worth anything to anyone but the CMO

**Comment by Gandalf2k15** [comment]
> Gandalf, what if instead of researching the virus cure, it spawns in the CMO's inventory or his locker as a high risk item that can either reveal the zombie cure tech or even make a zombie virus or something similar but more weaker, making finally a CMO high risk item worth anything to anyone but the CMO

I could totally add this quirk item actually

File: modular_skyrat/modules/mutants/code/mutant_species.dm (hunk 1 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/mutants/code/mutant_species.dm]: Yee, I've taken that into account and rebalanced them, they come in two types, fast mutants and slow ones.

**Comment by Owai-Seek** [comment]
I just want to point out that the first round this was tested, 90% of the station was converted within 10m.

As Elli said, they were extremely difficult to kill.

Their damage is super high. People were going down in just a few hits Getting tagged a single time meant you were infected. Getting tagged more than once meant you were dead. (Hell, even once was bad, because you get slow downed by damage.). Attempting to help anyone who was attacked just meant you joined the horde.

They can only be stopped by walls, forcefields, and blast doors. Everything else stands zero chance.

People who got converted started sabotaging stuff despite their brain being mush. Atmos, the engine, communication consoles, flooding, ect.

I saw I think three people infected who probably shouldn't have been. Someone in a security hardsuit, someone in an ERT hardsuit, and someone in a virologist suit.

**Comment by Trubbl3** [comment]
> > Gandalf, what if instead of researching the virus cure, it spawns in the CMO's inventory or his locker as a high risk item that can either reveal the zombie cure tech or even make a zombie virus or something similar but more weaker, making finally a CMO high risk item worth anything to anyone but the CMO
> 
> I could totally add this quirk item actually

I just want to give traitors or antags in general an incentive to go after the CMO because now theres no real value on going after the CMO unlike the other heads of staff

**Comment by H0gberg** [comment]
> I just want to point out that the first round this was tested, 90% of the station was converted within 10m.
> 
> As Elli said, they were extremely difficult to kill.
> 
> Their damage is super high. People were going down in just a few hits Getting tagged a single time meant you were infected. Getting tagged more than once meant you were dead. (Hell, even once was bad, because you get slow downed by damage.). Attempting to help anyone who was attacked just meant you joined the horde.
> 
> They can only be stopped by walls, forcefields, and blast doors. Everything else stands zero chance.
> 
> People who got converted started sabotaging stuff despite their brain being mush. Atmos, the engine, communication consoles, flooding, ect.
> 
> I saw I think three people infected who probably shouldn't have been. Someone in a security hardsuit, someone in an ERT hardsuit, and someone in a virologist suit.

I think it'd be smart if we just hard locked some stuff like hacking, or using guns much like how when you're hulked you can't use guns, it'd make them much more manageable and zombie like

**Comment by GoldenAlpharex** [comment]
The advanced RNA extractor is for some reason a normal-sized item, and its name is wrong in the protolathe, it's still called an Empty RNA Vial.

**Comment by Sigmar-alkahest** [comment]
holy line changes batman.

**Comment by Gandalf2k15** [comment]
Made this far less annoying to deal with, Cures now full revive people, as they should. It REALLY sucks being cured and having to wait for medical to do their shit. The cure is the real deal.
